src/dbnode/storage/flush.go

```diff
@@ -361,18 +430,30 @@ func (m *flushManager) flushNamespaceWithTimes(
 	ns databaseNamespace,
 	times []xtime.UnixNano,
 	flushPreparer persist.FlushPreparer,
-) error {
+) (namespaceFlush, error) {
+	flushes := make(map[shardFlush]bool, 0)
 	multiErr := xerrors.NewMultiError()
 	for _, t := range times {
 		// NB(xichen): we still want to proceed if a namespace fails to flush its data.
 		// Probably want to emit a counter here, but for now just log it.
-		if err := ns.WarmFlush(t, flushPreparer); err != nil {
+		shards, err := ns.WarmFlush(t, flushPreparer)
+		if err != nil {
 			detailedErr := fmt.Errorf("namespace %s failed to flush data: %v",
 				ns.ID().String(), err)
 			multiErr = multiErr.Add(detailedErr)
+		} else {
+			for _, s := range shards {
+				flushes[shardFlush{
+					shard: s,
+					time:  t,
+				}] = true
+			}
```

Comment:
nit: For more idiomatic Go better to do a continue above so you don't need an else here:
```go
		shards, err := ns.WarmFlush(t, flushPreparer)
		if err != nil {
			detailedErr := fmt.Errorf("namespace %s failed to flush data: %v",
				ns.ID().String(), err)
			multiErr = multiErr.Add(detailedErr)
			continue
		}

		for _, s := range shards {
			flushes[shardFlush{
				shard: s,
				time:  t,
			}] = true
		}
```

As per Effective Go:
https://golang.org/doc/effective_go#if
> In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.